### ISSUE TITLE
Failed to build Ubuntu16.04 image

### ISSUE THREAD

**Issue by zuojihong** [microsoft/azure-pipelines-image-generation#952]
When building Linux image using packer 1.3.4, it still reported failures when building python 2.7. 
We configure the env variables with shell script as below:

#!/bin/bash
export COMMIT_ID=ubuntu1604_agent_image_20190517
export ARM_CLIENT_ID=xxxxxxxx-xxxx-xxxx-xxxx-xxxxdxxxxxxx
export ARM_CLIENT_SECRET=xxxxxxxx-xxxx-xxxx-xxxx-xxxxdxxxxxxx
export ARM_TENANT_ID=xxxxxxxx-xxxx-xxxx-xxxx-xxxxdxxxxxxx
export ARM_SUBSCRIPTION_ID=688b835f-xxxx-xxxx-xxxx-xxxxxxxxxxx
export ARM_RESOURCE_GROUP=my-devops-rg
export ARM_STORAGE_ACCOUNT=mystore
export ARM_RESOURCE_LOCATION="Southeast Asia"

Below are the exception output:

    azure-arm:   inflating: lib/python2.7/hmac.pyo
    azure-arm:   inflating: lib/python2.7/csv.py
    azure-arm:   inflating: lib/python2.7/BaseHTTPServer.pyo
    azure-arm:   inflating: lib/python2.7/_strptime.py
    azure-arm:   inflating: lib/python2.7/sre_compile.py
    azure-arm:   inflating: lib/python2.7/uu.py
    azure-arm:   inflating: lib/python2.7/UserDict.pyo
    azure-arm:   inflating: lib/python2.7/tempfile.pyc
    azure-arm:   inflating: lib/python2.7/dis.py
    azure-arm:   inflating: lib/python2.7/HTMLParser.pyc
    azure-arm:   inflating: lib/python2.7/rfc822.py
    azure-arm:    creating: lib/pkgconfig/
    azure-arm:   inflating: lib/pkgconfig/python2.pc
    azure-arm:   inflating: lib/pkgconfig/python.pc
    azure-arm:   inflating: lib/pkgconfig/python-2.7.pc
    azure-arm:   inflating: lib/libpython2.7.a
    azure-arm:    creating: share/
    azure-arm:    creating: share/man/
    azure-arm:    creating: share/man/man1/
    azure-arm:   inflating: share/man/man1/python.1
    azure-arm:   inflating: share/man/man1/python2.7.1
    azure-arm:   inflating: share/man/man1/python2.1
    azure-arm:   inflating: python-2.7.16.tgz
==> azure-arm:
==> azure-arm: Cleanup requested, deleting resource group ...
==> azure-arm: Resource group has been deleted.
Build 'azure-arm' errored: Script exited with non-zero exit status: 1

==> Some builds didn't complete successfully and had errors:
--> azure-arm: Script exited with non-zero exit status: 1

==> Builds finished but no artifacts were created.

**Comment by zuojihong** [comment]
Did anyone else run into the same issue when building Ubuntu16.04 image?

**Comment by nelisbijl** [comment]
Not sure if directly related (also crashing at Python stage):

```
1558563051,,ui,message,    azure-arm: Python versions folders: 2.7.16 3.6.8 3.5.7 3.4.10 3.7.3
1558563051,,ui,message,    azure-arm: ------------------------------------------
1558563051,,ui,message,    azure-arm: Test /opt/hostedtoolcache/Python/2.7.16:
1558563051,,ui,error,==> azure-arm: /tmp/script_758.sh: line 26: /opt/hostedtoolcache/Python/2.7.16/x64/python: No such file or directory
1558563051,,ui,error,==> azure-arm: Script exited with non-zero exit status: 1.Allowed exit codes are: [0]
1558563051,,ui,error,==> azure-arm: Step "StepProvision" failed%!(PACKER_COMMA) aborting...
1558563051,,ui,error,Build 'azure-arm' errored: unexpected EOF
1558563051,,error-count,1
1558563051,,ui,error,\n==> Some builds didn't complete successfully and had errors:
1558563051,azure-arm,error,unexpected EOF
1558563051,,ui,error,--> azure-arm: unexpected EOF
1558563051,,ui,say,\n==> Builds finished but no artifacts were created.
```
I used on-error=abort; on the VM:

```
packer@pkrvms65s6gl7ch:~$ ls -al /opt/hostedtoolcache/Python/2.7.16/x64
total 17052
drwxr-xr-x 6 root root     4096 May 22 22:08 .
drwxr-xr-x 3 root root     4096 May 22 22:07 ..
drwxr-xr-x 2 root root     4096 May 22 22:08 bin
drwxr-xr-x 3 root root     4096 May 17 14:01 include
drwxr-xr-x 4 root root     4096 May 17 14:01 lib
-rw-r--r-- 1 root root 17431748 May 17 13:43 python-2.7.16.tgz
-rwxr-xr-x 1 root root      208 May 22 22:07 setup.sh
drwxr-xr-x 3 root root     4096 May 17 14:01 share
packer@pkrvms65s6gl7ch:~$
```

**Comment by alepauly** [comment]
@nelisbijl The issue you mentioned was fixed in #968 

@zuojihong yours looks like a different issue. Is there any more output that displays the actual error from the script?
